### ISSUE TITLE
[Bug] Markdown

### ISSUE THREAD

**Issue by TimerWolf** [go-gitea/gitea#10326]
- Gitea version: 1.10.0
- Git version: 2.17.1
- Operating system: Linux / Ubuntu 18.04.3 LTS / Bionic
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
  - [x] Not relevant
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (https://try.gitea.io/TimerWolf/MarkdownBug/src/branch/master/Markdown.md)
  - [ ] No
  - [ ] Not relevant
- Log gist:
  - [x] Not relevant

## Description

I think that this and the broken feature is called markdown, and if you use it this way as i do under this text it will break, for some reason it don't here at github :)

If the link provided not working, just use the code that i used here!

## Example

| # | Status | Hover
| -------- | -------- | -------- 
| 1 | Works | [Here](https://: "1, 2 ,3, 4, 5") |
| 2 | Broken | [Here](https://: "1 -> 2 -> 3 -> 4 -> 5") |

**Comment by jolheiser** [comment]
The markdown parser we use must not allow `>` in link titles. (`<` also aren't allowed, nor are most symbols from what I could tell)  
My best guess is it's to block users from breaking out of the rendered HTML tag and insert potentially malicious elements.

**Comment by zeripath** [comment]
Have you tried &gt;?

**Comment by lafriks** [comment]
To be honest I don't see difference between github and gitea rendered tables :thinking:

**Comment by TimerWolf** [comment]
> 
> 
> The markdown parser we use must not allow `>` in link titles. (`<` also aren't allowed, nor are most symbols from what I could tell)
> My best guess is it's to block users from breaking out of the rendered HTML tag and insert potentially malicious elements.

Maybe that's the case but why would it be allowed here on github then?

> 
> 
> To be honest I don't see difference between github and gitea rendered tables 🤔

The broken one don´t show you the tooltip in "try.gitea.io" when you hover over it.

**Comment by jolheiser** [comment]
GH uses a different parser than us, more than likely. I haven't had a chance to look for a possible config option, but my guess is we would need to extend our parser for this to work.

**Comment by guillep2k** [comment]
It took me a while to understand the problem 😁. For reference:

#### GitHub, mouse hover on first item:
![image](https://user-images.githubusercontent.com/18600385/74762260-3fdae380-525c-11ea-991b-b157b9ac514d.png)

#### GitHub, mouse hover on second item:
![image](https://user-images.githubusercontent.com/18600385/74762304-55500d80-525c-11ea-9503-6af49a103fdc.png)

#### Gitea, mouse hover on first item:
![image](https://user-images.githubusercontent.com/18600385/74762341-6bf66480-525c-11ea-85f2-9b535e39a0d6.png)

#### Gitea, mouse hover on second item:
![image](https://user-images.githubusercontent.com/18600385/74762372-7add1700-525c-11ea-89e6-5b7f8a5f70d4.png)

@TimerWolf since we can't access the exact markdown code you've used, would you paste it here in a code block?

```
    ```
    ### Markdown code
    [text](url)
    ```
```

**Comment by jolheiser** [comment]
@guillep2k 

```markdown
[Works](https://: "1, 2 ,3, 4, 5")
[Broken](https://: "1&lt;  2 -> 3 -> 4 -> 5")
```

**Comment by guillep2k** [comment]
No luck. The parser won't even take `＜` (Unicode U+FF1C - FULLWIDTH LESS-THAN SIGN). As far as I can tell, no symbol outside a small subset will be accepted. `*`, `$` and `#` are rejected, accented characters (`á`) and comma (`,`) seem OK, but colon, semicolon and ampersand are out of the question.

**Comment by jolheiser** [comment]
This looks like an upstream bug, possibly?
It should be allowed per [CommonMark spec](https://spec.commonmark.org/0.29/#example-502).

**Comment by TimerWolf** [comment]
@guillep2k: thanks for straight that things out for me, for some reason when i do stuff it always end up that people don't understand what i mean.

And i don´t know why as i was quite clear to me as i did change the code to "hover -> here" and also actually have the "status" indicator, but thanks again and sorry if anyone if it was a bit unclear, the code is back to its original i was never suppose to change not sure why i accepted that change...

**Comment by mrsdizzie** [comment]
This comes from default bluemonday policy and not goldmark:

https://github.com/microcosm-cc/bluemonday/blob/e02c01ee5f7fd95e94679ab2228aa83f5103b328/helpers.go#L109-L112

Then enabled for title here:

 https://github.com/microcosm-cc/bluemonday/blob/e02c01ee5f7fd95e94679ab2228aa83f5103b328/helpers.go#L162-L163

If we did want to overide it and allow a closing tag for link titles we could use this in https://github.com/go-gitea/gitea/blob/master/modules/markup/sanitizer.go, 

```
sanitizer.policy.AllowAttrs("title").Matching(regexp.MustCompile(`^[\p{L}\p{N}\s\-_',\[\]!\./\\\(\)\>]*$`)).OnElements("a")
```

Which in testing would output:

```
<td>Broken</td>
<td><a href="https://:" title="1 -&gt; 2 -&gt; 3 -&gt; 4 -&gt; 5">How</a></td>
```

Which seems fine in that example. I believe. You can see it currently doesn't allow other characters as well (pretty much anything that isn't a word number and those few punctuations listed)

**Comment by jolheiser** [comment]
Shoot, I thought I tested it well enough locally. Good catch @mrsdizzie!

**Comment by mrsdizzie** [comment]
@jolheiser  Does that seem like something we should change you think?

**Comment by jolheiser** [comment]
It seems that would solve this particular issue, but I'm not sure if there would be other unintended effects. Based on the comment in the bluemonday excerpt, it seems like `>` was left out intentionally.

If you test with that sanitizer, can you escape titles using raw html?

**Comment by mrsdizzie** [comment]
Hmm seems fine in my probably limited testing

**Comment by 6543** [comment]
We switcht from blackfriday to gildmark, so this may also fixed this issue

**Comment by jolheiser** [comment]
> We switcht from blackfriday to gildmark, so this may also fixed this issue

As mentioned above, this is because of our sanitizer, not goldmark.


> Hmm seems fine in my probably limited testing

@mrsdizzie would you mind opening a PR? We can discuss whether we should include more symbols there?

**Comment by mrsdizzie** [comment]
@jolheiser opened PR for further discussion

**Comment by lafriks** [comment]
Fixed by #10527 if not please reopen
